### PR TITLE
Introduce vanilla external platform

### PR DESCRIPTION
**Pull request by adriengentil**
- Introduce a new platform type for external, and its implementation as a provider
  - I kept the current behavior, `platformName` is not customizable by the user, and will be set to `external` in the instal-config 
- Introduce a new feature support level for it, behaves like `None` platform, the restrictions are:
  - Platform Managed Networking required (seems the new name for UMN?)
  - OCP 4.14 or greater is required
- We may want to put this new feature behind a capability for some time

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -3584,7 +3584,7 @@ paths:
           name: platform_type
           description: The provider platform type.
           type: string
-          enum: [ 'baremetal', 'none', 'nutanix', 'vsphere', 'oci' ]
+          enum: [ 'baremetal', 'none', 'nutanix', 'vsphere', 'oci', 'external' ]
```
Comment (avishayt) [inline swagger.yaml:3587]: Let's remove oci

File: swagger.yaml [hunk 2]
```diff
@@ -4086,6 +4086,7 @@ definitions:
       - 'EXTERNAL_PLATFORM_OCI'
       - 'DUAL_STACK'
       - 'PLATFORM_MANAGED_NETWORKING'
+      - 'EXTERNAL_PLATFORM'
```
Comment (avishayt) [inline swagger.yaml:4089]: Remove EXTERNAL_PLATFORM_OCI above

File: swagger.yaml [hunk 3]
```diff
@@ -6226,6 +6227,7 @@ definitions:
       - vsphere
       - none
       - oci
+      - external
```
Comment (avishayt) [inline swagger.yaml:6230]: Remove oci above

File: swagger.yaml (hunk 1 above)
Comment (adriengentil) [reply swagger.yaml:3587]: don't we want to remove it once the UI has migrated to external+platform name?

Comment (avishayt) [reply swagger.yaml:3587]: Yes

**Comment by adriengentil** [comment]
/close
I'll re-open once we start to work on https://issues.redhat.com/browse/MGMT-15654

**Pull request closed without merging**